src/domain/user.js

```diff
@@ -236,6 +236,61 @@ export default class User {
     })
     return updatedUser
   }
+
+  static async searchUserByName(name) {
```

Comment:
Since this is a domain function, let's stick to the standard of `findByName` instead of search